File: 프로그래머스/Level 2/택배상자-ex.java
import java.util.Stack;
import java.util.Queue;
import java.util.LinkedList;

class Solution {
    public int solution(int[] order) {
        int loadedBoxes = 0;
        int orderIdx = 0;
        
        Stack<Integer> auxBelt = new Stack<>();
        Queue<Integer> belt = new LinkedList<>();
        
        for (int i = 0; i < order.length; i++) {
            auxBelt.add(i + 1);
            while (!auxBelt.isEmpty()) {
                if (auxBelt.peek() == order[orderIdx]) {
                    belt.offer(auxBelt.pop());
                    orderIdx++;
                } else break;
            }
        }
        
        loadedBoxes = belt.size();
        return loadedBoxes;
    }
}

File: 프로그래머스/Level 2/택배상자-ex.py
def solution(order):
    answer = 0
    stacks = [] 
    n = len(order)
    i = 1
    idx = 0
    
    while i < n + 1:
        stacks.append(i)
        while stacks[-1] == order[idx]:
            idx += 1
            stacks.pop()
            if len(stacks) == 0:
                break
                
        i += 1
        
    return idx